modules/heat_conduction/test/tests/recover/tests

```diff
@@ -22,6 +24,47 @@
     delete_output_before_running = false
     cli_args = '--recover recover_out_cp/0005'
     superlu = true
-    skip = 'Non-deterministic on Intel compiler #10737'
+    requirement = 'MOOSE shall be able to recover from a short simulation and reproduce a the full time scale simulation with heat conduction, a heat source, thermal contact, and boundary conditions.'
+  [../]
+
+  [./ad_recover_1]
+    type = Exodiff
+    input = ad_recover.i
+    exodiff = ad_recover_out.e
+    superlu = true
+    requirement = 'MOOSE shall run a simulation with heat conduction, a heat source, thermal contact, and boundary conditions with automatic differentiation.'
+  [../]
+  [./ad_recover_2]
+    type = RunApp
+    input = ad_recover.i
+    prereq = ad_recover_1
+    cli_args = 'Outputs/checkpoint=true Executioner/num_steps=5'
+    superlu = true
+    requirement = 'MOOSE shall run a short simulation with heat conduction, a heat source, thermal contact, and boundary conditions with automatic differentiation.'
+  [../]
+  [./ad_recover_3]
+    type = Exodiff
+    input = ad_recover.i
+    exodiff = ad_recover_out.e
+    prereq = ad_recover_2
+    delete_output_before_running = false
+    cli_args = '--recover ad_recover_out_cp/0005'
+    superlu = true
+    requirement = 'MOOSE shall be able to recover from a short simulation and reproduce a the full time scale simulation with heat conduction, a heat source, thermal contact, and boundary conditions with automatic differentiation.'
+  [../]
+
+  [./ad_recover_jacobian]
+    type = 'PetscJacobianTester'
+    input = 'ad_recover.i'
+    ratio_tol = 1e-7
+    difference_tol = 1e-5
+    run_sim = True
+    petsc_version = '>=3.9'
+    method = 'OPT'
+    cli_args = 'Executioner/num_steps=2 Outputs/exodus=false'
+    skip = 'AD Thermal Contact not yet implemented'
```

Comment:
I like to envision that we will be able to do this by the end of the fiscal year

Reply:
I was looking through it, and I'm not sure how to handle the aux pieces. Quadrature thermal contact touches:
* `GapValueAux`
* `PenetrationAux`
* `GapHeatTransfer` boundary condition
* `GapConductance` material
* `PenetrationLocator`

Either the aux's don't have jacobian information (but I think they do since it's telling how far apart the pieces are) or they have to be converted to materials before going to AD materials. I'm not sure if they _have_ to be aux's or not. I'm willing to help get this in earlier than FY, since this is the last missing piece to do fully AD thermal solves in bison @bwspenc @dschwen